### ISSUE TITLE
xenstored uses 100% CPU after updates

### ISSUE THREAD

**Issue by arjan-s** [QubesOS/qubes-issues#4191]
Qubes OS version: 4.0

After running dom0 updates on August 6th and a reboot, the process xenstored takes 100% of one CPU core. Another reboot does not change this.
I traced this issue to the upgrade of xen-hvm-stubdom-linux to version 1.0.8-1.fc25 by selectively downgrading it and rebooting.

**Comment by lunarthegrey** [comment]
Strange. I am not experiencing this with the latest stable updates on 4.0. Would an `strace -tfp` of the PID provide any valuable information to debuggers?

**Comment by seberm** [comment]
I observe same problem :/.

$ ps -aux | grep xenstored
`root      1860 77.5  0.0  47956  3572 ?        RL   09:52 184:16 /usr/sbin/xenstored --pid-file /var/run/xen/xenstored.pid`

$ strace -tfp 1860
```
...
13:46:58 read(22, "<\373\5\0", 4)       = 4
13:46:58 lseek(22, 391996, SEEK_SET)    = 391996
13:46:58 read(22, "\270\364\5\0H\0\0\0&\0\0\0\35\0\0\0\254\256I3\231\31\1&", 24) = 24
13:46:58 lseek(22, 392020, SEEK_SET)    = 392020
13:46:58 read(22, "/local/domain/6/backend/vif/13/0"..., 38) = 38
13:46:58 lseek(22, 392058, SEEK_SET)    = 392058
13:46:58 read(22, "\2\0\0\0\1\0\0\0\0\0\0\0\6\0\0\0\0\0\0\0\r\0\0\0\1\0\0\0006", 29) = 29
13:46:58 fcntl(22, F_SETLKW, {l_type=F_UNLCK, l_whence=SEEK_SET, l_start=16800, l_len=1}) = 0
13:46:58 ioctl(12, IOCTL_EVTCHN_NOTIFY, 0x7ffdc71c2f7c) = 0
13:46:58 ioctl(12, IOCTL_EVTCHN_NOTIFY, 0x7ffdc71c2f7c) = 0
13:46:58 poll([{fd=3, events=POLLIN|POLLPRI}, {fd=4, events=POLLIN|POLLPRI}, {fd=5, events=POLLIN|POLLPRI}, {fd=12, events=POLLIN|POLLPRI}, {fd=13, events=POLLIN|POLLPRI}, {fd=14, events=POLLIN|POLLPRI}, {fd=15, events=POLLIN|POLLPRI}, {fd=16, events=POLLIN|POLLPRI}, {fd=17, events=POLLIN|POLLPRI}, {fd=18, events=POLLIN|POLLPRI}, {fd=23, events=POLLIN|POLLPRI}, {fd=8, events=POLLIN|POLLPRI}, {fd=19, events=POLLIN|POLLPRI}], 13, 0) = 1 ([{fd=12, revents=POLLIN}])
13:46:58 read(12, "\204\0\0\0", 4)      = 4
13:46:58 write(12, "\204\0\0\0", 4)     = 4
13:46:58 ioctl(12, IOCTL_EVTCHN_NOTIFY, 0x7ffdc71c2f3c) = 0
13:46:58 ioctl(12, IOCTL_EVTCHN_NOTIFY, 0x7ffdc71c2f3c) = 0
13:46:58 fcntl(22, F_SETLKW, {l_type=F_RDLCK, l_whence=SEEK_SET, l_start=16800, l_len=1}) = 0
13:46:58 lseek(22, 16804, SEEK_SET)     = 16804
13:46:58 read(22, "<\373\5\0", 4)       = 4
13:46:58 lseek(22, 391996, SEEK_SET)    = 391996
13:46:58 read(22, "\270\364\5\0H\0\0\0&\0\0\0\35\0\0\0\254\256I3\231\31\1&", 24) = 24
13:46:58 lseek(22, 392020, SEEK_SET)    = 392020
13:46:58 read(22, "/local/domain/6/backend/vif/13/0"..., 38) = 38
13:46:58 lseek(22, 392058, SEEK_SET)    = 392058
13:46:58 read(22, "\2\0\0\0\1\0\0\0\0\0\0\0\6\0\0\0\0\0\0\0\r\0\0\0\1\0\0\0006", 29) = 29
13:46:58 fcntl(22, F_SETLKW, {l_type=F_UNLCK, l_whence=SEEK_SET, l_start=16800, l_len=1}) = 0
13:46:58 ioctl(12, IOCTL_EVTCHN_NOTIFY, 0x7ffdc71c2f7c) = 0
13:46:58 ioctl(12, IOCTL_EVTCHN_NOTIFY, 0x7ffdc71c2f7c) = 0
13:46:58 poll([{fd=3, events=POLLIN|POLLPRI}, {fd=4, events=POLLIN|POLLPRI}, {fd=5, events=POLLIN|POLLPRI}, {fd=12, events=POLLIN|POLLPRI}, {fd=13, events=POLLIN|POLLPRI}, {fd=14, events=POLLIN|POLLPRI}, {fd=15, events=POLLIN|POLLPRI}, {fd=16, events=POLLIN|POLLPRI}, {fd=17, events=POLLIN|POLLPRI}, {fd=18, events=POLLIN|POLLPRI}, {fd=23, events=POLLIN|POLLPRI}, {fd=8, events=POLLIN|POLLPRI}, {fd=19, events=POLLIN|POLLPRI}], 13, 0) = 1 ([{fd=12, revents=POLLIN}])
13:46:58 read(12, "\204\0\0\0", 4)      = 4
13:46:58 write(12, "\204\0\0\0", 4)     = 4
13:46:58 ioctl(12, IOCTL_EVTCHN_NOTIFY, 0x7ffdc71c2f3c) = 0
13:46:58 ioctl(12, IOCTL_EVTCHN_NOTIFY, 0x7ffdc71c2f3c) = 0
13:46:58 fcntl(22, F_SETLKW, {l_type=F_RDLCK, l_whence=SEEK_SET, l_start=16800, l_len=1}) = 0
13:46:58 lseek(22, 16804, SEEK_SET)     = 16804
....
```

And this output still repeats ...

@arjan-s could you please provide exact steps how did you downgrade? This problem makes my system unusable.

**Comment by arjan-s** [comment]
Thanks, due to depending on my laptop for work I didn't have the possibility yet to create an strace.

Downgrading is easy:
```
[user@dom0 ~]$ sudo qubes-dom0-update --action=downgrade xen-hvm-stubdom-linux
```

**Comment by taradiddles** [comment]
Thanks for the fix above.

Interestingly, xen-hvm-stubdom-linux was updated 2-3 weeks ago on my laptop but I witnessed the high CPU issue only twice, which I "fixed" with reboots.
After the L1TP updates this morning, xenstored was alway stuck at 80% and rebooting didn't help (tried 3 times in a row). Strace looked like @seberm 's. Downgrading xen-hvm-stubdom-linux did the trick. FYI, if downgrading and rebooting is not possible, one can stop/resume xenstored after/before qvm operations with `killall -SIGSTOP xenstored` and `killall -SIGCONT xenstored`
